apps/homescreen/js/pages.js

```diff
@@ -219,9 +219,10 @@
       // Remove child immediately, datastore operations can be quite slow
       this.removeCard(this.selectedCard);
 
+      var revId = this.pagesStore.datastore.revisionId;
       this.pagesStore.get(id).then(entry => {
         entry.data.pinned = false;
-        this.pagesStore.datastore.put(entry.data, id).then(() => {},
+        this.pagesStore.datastore.put(entry.data, id, revId).then(() => {},
```

Comment:
Firefox Sync updates each record at most once per hour, and only if you visited that same URL with your desktop browser during that last hour. The update should be fast, so you would have to aim very well to hit that time window with your unpin action. I think for most users it will happen zero times in the life time of their phone.